### PR TITLE
V3a ui uncruftify

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR contains the most recent UI and State and RPC for alpha demo app. 

This PR includes:

- [x] Chore work
- [ ] Bug fixes
- [x] Backwards-compatible feature additions
- [ ] Breaking changes
- [ ] Documentation updates
- [x] Test updates

**Note**: If you did not check "Documentation updates" and/or "Test updates" then your PR may not be ready!

## changelog

Write descriptions for your changes. If a change is related to a GitHub issue, please tag the issue in the description.

- (Chore) Screens to match designs
- (Fix) Change description
- (Feature) Full Calibration Experience Alpha
- (Refactor) State and Style refactors as UX flow was introduced
- (Docs) Change description
- (Tests) Corresponding selector and reducer tests

**Note**: If your PR incorporates many changes, or many different types of changes, please consider splitting it up into smaller PRs.

## review requests
@mcous can you take a look at the app merge conflict fixes
@astaff can you confirm api merge conflict resolutions use your latest code

### CODE REVIEW

File: app/ui/robot/actions.js [hunk 1]
```diff
@@ -213,7 +222,11 @@ export const actions = {
 
   // TODO(mc, 2017-10-04): make this action FSA compliant (error [=] bool)
   cancelResponse (error) {
-    return {type: actionTypes.CANCEL_RESPONSE, error}
+    const didError = error != null
```
Comment (astaff) [inline app/ui/robot/actions.js]: I see this being repeated 6 times with action type being the only variable. Wrap into a factory maybe?

File: app/ui/robot/api-client/client.js [hunk 2]
```diff
@@ -115,24 +119,25 @@ export default function client (dispatch) {
 
   function probeTip (state, action) {
     const {payload: {instrument: axis}} = action
-    const instrument = selectors.getState(state).protocolInstrumentsByAxis[axis]
+    const instrument = selectors.getInstrumentsByAxis(state)[axis]
 
     // FIXME(mc, 2017-10-05): DEBUG CODE
-    // return setTimeout(() => dispatch(actions.probeTipResponse()), 2000)
-
+    // return setTimeout(() => dispatch(actions.probeTipResponse()), 1000)
```
Comment (astaff) [inline app/ui/robot/api-client/client.js]: Do we still need that?

File: app/ui/robot/api-client/client.js [hunk 3]
```diff
@@ -240,20 +241,19 @@ export default function client (dispatch) {
     }
 
     // TODO(mc, 2017-08-30): Use a reduce
-    commands.forEach(makeHandleCommand())
-    instruments.forEach(apiInstrumentToInstrument)
-    containers.forEach(apiContainerToContainer)
+    ;(commands || []).forEach(makeHandleCommand())
```
Comment (astaff) [inline app/ui/robot/api-client/client.js]: Just curious: when is it a good practice to use semicolon this way

File: app/ui/robot/reducer/session.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  return {...state, resumeRequest: {inProgress: false, error: null}}
+}
+
+function handleCancel (state, action) {
+  return {...state, cancelRequest: {inProgress: true, error: null}}
+}
+
+function handleCancelResponse (state, action) {
```
Comment (astaff) [inline app/ui/robot/reducer/session.js]: Seems like a lot of boilerplate to me. Would this code be a good candidate to use https://github.com/reduxactions/redux-actions ?

File: app/ui/robot/api-client/client.js (hunk 2 above)
Comment (Kadee80) [reply app/ui/robot/api-client/client.js]: We need it for testing screens with virtual smoothie

File: app/ui/robot/index.js [hunk 5]
```diff
@@ -1,12 +1,10 @@
 // robot state module
 // split up into reducer.js, action.js, etc if / when necessary
```
Comment (mcous) [inline app/ui/robot/index.js]: TODO has been addressed

File: app/ui/robot/actions.js (hunk 1 above)
Comment (mcous) [reply app/ui/robot/actions.js]: See comment below about Redux boilerplate

**Comment by mcous** [comment]
Will likely be closing this PR in favor of just moving forward with branch `v3a` (which includes most, if not all, of the changes here)

File: app/ui/robot/reducer/session.js (hunk 4 above)
Comment (astaff) [reply app/ui/robot/reducer/session.js]: Thanks!

**Comment by mcous** [comment]
Closing this PR because we probably won't need it. Will keep the branch around for a little just in case

**Pull request closed without merging**